design/deletion-plugins.md

```diff
@@ -0,0 +1,80 @@
+# Deletion Plugins
+
+
+## Abstract
+Velero should introduce a new type of plugin that runs when a backup is deleted.
+These plugins will delete any external resources associated with the backup so that they will not be left orphaned.
+
+## Background
+With the CSI plugin, Velero developers introduced a pattern of using BackupItemAction and RestoreItemAction plugins tied to PersistentVolumeClaims to create other resources to complete a backup.
+In the CSI plugin case, Velero does clean up of these other resources, which are Kubernetes Custom Resources, within the core Velero server.
+However, for external plugins that wish to use this same pattern, this is not a practical solution.
+Velero's core cannot be extended for all possible Custom Resources, and not external resources that get created are Kubernetes Custom Resources.
+
+Therefore, Velero needs some mechanism that allows plugin authors who have created resources within a BackupItemAction or RestoreItemAction plugin to ensure those resources are deleted, regardless of what system those resources reside in.
+
+## Goals
+- Provide a new plugin type in Velero that is invoked when a backup is deleted.
+
+## Non Goals
+- Implementations of specific deletion plugins.
+- Rollback of deletion plugin execution.
+
+
+## High-Level Design
+Velero will provide a new plugin type that is similar to its existing plugin architecture.
+These plugins will be referred to as `DeleteAction` plugins.
+`DeleteAction` plugins will receive the `Backup` CustomResource being deleted on execution.
```

Comment:
@dsu-igeek I've added a new proposal with DeleteItemAction.

DeleteAction was chosen at first because don't currently download the tarball and iterate over it - we just delete the backup tarball entirely, since we're deleting the _backup_, not individual resources in it.

Reply:
The DeleteItemAction looks like what we need!

Reply:
@dsu-igeek Thanks!